src/Processors/Transforms/buildPushingToViewsChain.cpp

```diff
@@ -1,1032 +0,0 @@
-{
-    if (output.isFinished())
-        throw Exception(ErrorCodes::LOGICAL_ERROR, "Cannot finalize views because output port is finished");
-
-    if (!output.canPush())
-        return Status::PortFull;
-
-    bool ignore_errors = views_data->context->getSettingsRef()[Setting::materialized_views_ignore_errors];
-    size_t num_finished = 0;
-    size_t pos = 0;
-    for (auto & input : inputs)
-    {
-        auto i = pos;
-        ++pos;
-
-        if (input.isFinished())
-        {
-            ++num_finished;
-            continue;
-        }
-
-        input.setNeeded();
-        if (input.hasData())
-        {
-            auto data = input.pullData();
-            //std::cerr << "********** FinalizingViewsTransform got input " << i << " has exc " << bool(data.exception) << std::endl;
-            if (data.exception)
-            {
-                if (views_data->has_exception && views_data->first_exception == data.exception)
-                    statuses[i].is_first = true;
-                else
-                    statuses[i].exception = data.exception;
-
-                if (i == 0 && statuses[0].is_first && !ignore_errors)
-                {
-                    output.pushData(std::move(data));
-                    return Status::PortFull;
-                }
-            }
-
-            if (input.isFinished())
-                ++num_finished;
-        }
-    }
-
-    if (num_finished == inputs.size())
-    {
-        if (!statuses.empty())
-            return Status::Ready;
-
-        if (any_exception && !ignore_errors)
-            output.pushException(any_exception);
-
-        output.finish();
-        return Status::Finished;
-    }
-    return Status::NeedData;
-}
-
-static std::exception_ptr addStorageToException(std::exception_ptr ptr, const StorageID & storage)
-{
-    try
-    {
-        std::rethrow_exception(ptr);
-    }
-    catch (DB::Exception & exception)
-    {
-        exception.addMessage("while pushing to view {}", storage.getNameForLogs());
-        return std::current_exception();
-    }
-    catch (...)
-    {
-        return std::current_exception();
-    }
-}
-
-void FinalizingViewsTransform::work()
-{
-    size_t i = 0;
-    for (auto & view : views_data->views)
-    {
-        auto & status = statuses[i];
-        ++i;
-
-        if (status.exception)
-        {
-            if (!any_exception)
-                any_exception = status.exception;
-
-            view.setException(addStorageToException(status.exception, view.table_id));
-
-            /// Exception will be ignored, it is saved here for the system.query_views_log
-            if (views_data->context->getSettingsRef()[Setting::materialized_views_ignore_errors])
-                tryLogException(view.exception, getLogger("PushingToViews"), "Cannot push to the storage, ignoring the error");
-        }
-        else
-        {
-            view.runtime_stats->setStatus(QueryViewsLogElement::ViewStatus::QUERY_FINISH);
-
-            LOG_TRACE(
```

Comment:
Hi @CheSema,
I see that some users find this log useful for various debugging purposes. Do I understand correctly that it was removed prior to the `view_duration_ms` in `system.query_views_log`, which shows exactly the same value?